### ISSUE TITLE
[Discussion] How does the validation work?

### ISSUE THREAD

**Issue by JavaShipped** [deezer/spleeter#371]
Hi all, 

I've decided to train my own model, but just need some clarification on the train.csv vs. validation.csv. 

They have different files in it and the validation is significantly smaller. 

Is Spleeter training a model using the train.csv and then taking some other, similar (not the same) data/music and comparing it? If so, this means that none of the data can be shared between the two csv files, correct? 

So as a simple example, I could use the first 3 Foo Fighters Albums to train a model, and their most recent one to validate?

**Comment by dankwartrustow** [comment]
I can only comment on your third question:  A model’s ability to recognize the individual patterns in the albums from its dataset will be governed by how similar or dissimilar the new information it is receiving is.  In other words, music that has been mixed similarly will produce the best results. If you check the production information and find variations between albums, it could be because they were using different production houses and equalized their records differently. Reading up on how equalization works could help you in your task – but I’d be interested to know how this works out for you.   From: JavaShippedSent: Wednesday, May 13, 2020 9:32 AMTo: deezer/spleeterCc: SubscribedSubject: [deezer/spleeter] [Discussion] How does the validation work? (#371) Hi all,I've decided to train my own model, but just need some clarification on the train.csv vs. validation.csv.They have different files in it and the validation is significantly smaller.Is Spleeter training a model using the train.csv and then taking some other, similar (not the same) data/music and comparing it? If so, this means that none of the data can be shared between the two csv files, correct?So as a simple example, I could use the first 3 Foo Fighters Albums to train a model, and their most recent one to validate?—You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by romi1502** [comment]
Hi @JavaShipped 
> Is Spleeter training a model using the train.csv and then taking some other, similar (not the same) data/music and comparing it?

Spleeter does train the model using the train.csv and then regularly computes some metrics on the validation.csv. The metrics on the validation.csv aim at monitoring the ability of the model to generalize to unseen data (so it has to be different data). Checking these metrics can help detecting possible overfitting of the model.
This is a very common setup in general machine learning.

> If so, this means that none of the data can be shared between the two csv files, correct?

Exactly, otherwise, you won't monitor the ability of your model to generalize to unseen data.

> So as a simple example, I could use the first 3 Foo Fighters Albums to train a model, and their most recent one to validate?

Well if your goal is only to separate Foo fighters songs in the end, it may be sufficient. 
But, if you want your model being able to generalize to other bands (with other singers, other musicians, recording in other studios...), you'll need to put other bands song in the validation set to monitor it.
So it is also usually recommended to ensure that no artists appears both in the training set and the validation dataset (and the same for the test set), to avoid over optimistic metrics. By the way, this is not only valid for source separation but for any machine learning task. See for instance "A Closer Look On Artist Filters For Musical Genre Classification". Arthur Flexer, ISMIR 2007.
